agimus_controller/agimus_controller/mpc.py

```diff
@@ -0,0 +1,39 @@
+import time
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryBuffer, TrajectoryPoint
+from agimus_controller.mpc_output import MPCOutputData, MPCDebugData
+
+
+class MPC(object):
+    def __init__(self) -> None:
+        self._ocp = None
+        self._warm_start = None
+        self._mpc_output_data = MPCOutputData()
+        self._mpc_debug_data = MPCDebugData()
+
+    def setup(self, ocp: OCPBase, warm_start: WarmStartBase) -> None:
+        self._ocp = ocp
+        self._warm_start = warm_start
+
+    def run(
+        self, initial_state: TrajectoryPoint, trajectory_buffer: TrajectoryBuffer
+    ) -> MPCOutputData:
+        assert self._ocp is not None
+        assert self._warm_start is not None
+        timer1 = time.perf_counter()
+        reference_trajectory = trajectory_buffer.get_points(
+            self._ocp.get_horizon_size()
+        )
+        self._ocp.update_horizon_from_reference(reference_trajectory)
```

Comment:
Maybe this should be called `set_reference_horizon` or `update_reference_horizon`? Current name at first for me sounded like this is warmstart and not reference. This is simple naming change, but maybe it will be clearer later on